docs/self-hosting/configuration-options.md

```diff
@@ -26,10 +26,165 @@ After changing any of the environment variables mentioned below you will have to
+
+- `SNS_TOPIC_ARN`: ARN of the SNS topic that the service will publish events to.
+- `SNS_AWS_REGION`: AWS region of the SNS topic.
+- `SQS_QUEUE_URL`: URL of the SQS queue from which a worker will consume events.
+- `SQS_AWS_REGION`: AWS region of the SQS queue.
+- `S3_BACKUP_BUCKET_NAME`: name of the S3 bucket on which a file backup will be performed to transfer large data between services.
+- `S3_AWS_REGION`: AWS region of the S3 bucket.
+
+### Extensions Server Redirection (Optional)
+
+We use internal service discovery endpoints in our Virtual Private Cloud. In order for the requests to not go out via the internet we reroute the traffic go through internal network only.
+
+- `INTERNAL_DNS_REROUTE_ENABLED`: if you'd like to redirect posting notes to extensions server to a different url you can enable this option.
+- `EXTENSIONS_SERVER_URL`: the url of the extensions server to which the notes will be redirected.
+
+### Auth Service
+
+We are in the process of moving some user/authentication related functionalities out of Syncing Server JS and into Auth serivce. In order to achieve that gradualy we are redirecting some requests internally to the Auth service from Syncing Server js to slowly strangle that functionality in Syncing Server JS.
+
+-`AUTH_SERVER_URL`: url to the Auth service. Default value should be kept.
+
 ### Disabling new user registrations
 
-1. Set the `DISABLE_USER_REGISTRATION` environment variable to `true` in your `.env` file
+This is one of the features that will be deprecated soon in favor of Auth service. For now it is configurable in both Syncing Server JS and Auth.
+
+- `DISABLE_USER_REGISTRATION`: disable the option to register new users on the server.
+
+### Emails
+
+- `EMAIL_ATTACHMENT_MAX_BYTE_SIZE`: Amount of bytes allowed for the attachment with a data backup.
+
+### Revisions
+
+- `REVISIONS_FREQUENCY`: Amount of seconds between each revision save of a note.
+
+### New Relic (Optional)
+
+We are utilizing New Relic to monitor our infrastructure. If you wish to set up your own monitoring in New Relic you can utilize the following environment variables:
+
+- `NEW_RELIC_ENABLED`: enable or disable new relic agent
+- `NEW_RELIC_APP_NAME`: name of the syncing server application under which it will show in New Relic
+- `NEW_RELIC_LICENSE_KEY`: New Relic license key
+- `NEW_RELIC_NO_CONFIG_FILE`: should be true as we do not use configuration files for New Relic and fallback to environment variables.
+- `NEW_RELIC_DISTRIBUTED_TRACING_ENABLED`: enable or disable distrubuted tracing.
+- `NEW_RELIC_LOG_ENABLED`: enable or disable logs in New Relic
+- `NEW_RELIC_LOG_LEVEL`: level of logs in New Relic
```

Comment:
Some descriptions seem to have a period at the end and some don't. I think for consistency we should just add a period after after description.